server/storage/generic.go

```diff
@@ -9,38 +9,57 @@ import (
 	"io"
 )
 
+// ObjectID is a Git Object ID (sha1 digest in hex format without prefix)
 type ObjectID string
 
+// ZeroID is the "null" object ID, indicating there's no object here
 const ZeroID ObjectID = "0000000000000000000000000000000000000000"
 
+// ObjectIDLength defines the length of an Object ID
 const ObjectIDLength = 40
 
+// ObjectIDFromRaw parses the bytes of a sha1 digest and returns the standard object ID format representation
 func ObjectIDFromRaw(rawid []byte) ObjectID {
 	return ObjectID(hex.EncodeToString(rawid))
 }
 
-type Reference string
-
+// ObjectType represents the type of a Git Object
+// In Pack files, this is represented by 3 bits.
 type ObjectType int
 
 const (
+	// ObjectTypeBad is an invalid object type and should result in an error
 	ObjectTypeBad    ObjectType = -1
+	// ObjectTypeNone should never occur, as it's an encoded value for invalid object
 	ObjectTypeNone   ObjectType = 0
+	// ObjectTypeCommit encodes a "Commit" object
 	ObjectTypeCommit ObjectType = 1
+	// ObjectTypeTree encodes a "tree" object
 	ObjectTypeTree   ObjectType = 2
+	// ObjectTypeBlob encodes a "blob" object
 	ObjectTypeBlob   ObjectType = 3
+	// ObjectTypeTag encodes a "tag" object
 	ObjectTypeTag    ObjectType = 4
+
 	// 5 is "for future expansion"
+
+	// ObjectTypeOfsDelta encodes an "offset delta" object
+	// These are deltas that refer to another object in the same pack by offset from the base
 	ObjectTypeOfsDelta ObjectType = 6
+	// ObjectTypeRefDelta encodes a "reference delta" object
+	// These are deltas that refer to another object in the same repository by encoding the base object ID
 	ObjectTypeRefDelta ObjectType = 7
+	// ObjectTypeAny should never occur, as it's an encoded value for "any" object (anything goes), which we don't want
 	ObjectTypeAny      ObjectType = 8
+	// ObjectTypeMax should never occur, as it's higher than available in the 3 bits reserved for object types
 	ObjectTypeMax      ObjectType = 9
```

Comment:
It's defined here because this set of definitions comes from https://github.com/git/git/blob/master/cache.h#L432 .
What I meant is that they should never occur returned from specific functions other than if their response value is documented as such (e.g. the `tree` driver sends it back from `ReadObject` when an object was not found).
But their values should *never* occur in external (pack)files, as doing such would be a clear error.